### PR TITLE
Clarify `PLATFORM=all` description

### PR DESCRIPTION
**Pull request by gervaisj**
This resolves https://github.com/adoptium/aqa-tests/issues/2227 by clarifying the description of the `all` platform, notably pointing out how some secondary platforms are excluded and the selected JDK version also changes the platforms.

The "2nd part" of the issue does not apply anymore (no `_XL` platforms are included). See https://github.com/adoptium/aqa-tests/issues/2227#issuecomment-1019375666 for more details.

### CODE REVIEW

**Comment by gervaisj** [comment]
@smlambert @llxia Ready for review

File: buildenv/jenkins/testJobTemplate [hunk 1]
```diff
@@ -279,7 +279,7 @@ ARCH_OS_LIST.each { ARCH_OS ->
                                 <br/>sparcv9_solaris
                                 <br/>riscv64_linux
                                 <br/>multiple platforms (comma separated values): x86-64_linux,x86-64_mac
-                                <br/>all platforms: all (actual platforms will change based on JDK_VERSION/JDK_IMPL)
+                                <br/>all platforms: all (this set actually excludes some secondary platforms and changes based on JDK_VERSION/JDK_IMPL selected)
```
Comment (smlambert) [inline buildenv/jenkins/testJobTemplate:282]:
```suggestion
                                <br/>all platforms: all (this set actually excludes some secondary platforms and varies based on JDK_VERSION/JDK_IMPL selected)
```

**Review by smlambert — APPROVED** [review]
minor word choice suggestion, LGTM otherwise, thanks @gervaisj !

**Review by Mesbah-Alam — APPROVED** [review]
LGTM. Approving based on assumption that the suggested change will be made before merge!

**Pull request merged**